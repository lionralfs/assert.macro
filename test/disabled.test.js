const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  pluginName: 'assert.macro - disabled',
  tests: [
    {
      title: 'Remove import if unused',
      code: `import assert from '../dist/assert.macro';`
    },
    {
      title: 'Remove import & references even if used',
      code: `
        import assert from '../dist/assert.macro';
        
        function doSomething(val) {
            assert(val > 0, 'Custom error message');
        }`
    },
    {
      title: 'Throws if anything other than the default export is used',
      code: `import assert, { nonExistentExport } from '../dist/assert.macro';`,
      error: function(error) {
        return error instanceof MacroError && /only import assert as default/.test(error.message) && /nonExistentExport/.test(error.message);
      }
    }
  ]
});
