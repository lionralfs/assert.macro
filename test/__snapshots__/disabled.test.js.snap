// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assert.macro - disabled Remove import & references even if used: Remove import & references even if used 1`] = `
"
import assert from '../dist/assert.macro';

function doSomething(val) {
    assert(val > 0, 'Custom error message');
}

      ↓ ↓ ↓ ↓ ↓ ↓

function doSomething(val) {}
"
`;

exports[`assert.macro - disabled Remove import if unused: Remove import if unused 1`] = `
"
import assert from '../dist/assert.macro';

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;

exports[`assert.macro - disabled Throws if anything other than the default export is used: Throws if anything other than the default export is used 1`] = `
"
import assert, { nonExistentExport } from '../dist/assert.macro';

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: You should only import assert as default. You are also importing nonExistentExport.
"
`;
